added adding new Player and error handling when providing a nickname

choooobert opened the pull request.

choooobert committed:
added connection to backend

choooobert committed:
added IP to user

choooobert committed:
added error handling when providing bad nickname

choooobert committed:
removed IP from Player

choooobert committed:
changed backend URL to master version

Kamil-Jankowski commented on src/app/home/home.component.html:
```diff
@@ -28,7 +28,7 @@ <h4 class="card-title">Ships: The Game</h4>
         <p class="bs-component">
             <a>
-                <button (click)="add(playerName.value)" type="button" class="btn btn-primary btn-lg btn-block" routerLink="/waiting-room">Join the game</button>
+                <button (click)="add(playerName.value)" type="button" class="btn btn-primary btn-lg btn-block">Join the game</button>
```
Dlaczego nie przechodzimy do poczekalni?

Kamil-Jankowski commented on src/app/player.service.ts:
```diff
@@ -1,41 +1,42 @@
 })
 export class PlayerService {
-  private playersUrl = 'api/players';  // URL to web api
+  private playersUrl = 'https://ships-room-service-backend.herokuapp.com/room';  // URL to web api
```
Nie można się polączyć przy uruchomieniu frontendu na _localhost_

Kamil-Jankowski reviewed (commented):
Brak połączenia z heroku przy lokalnym uruchomieniu frontendu uniemożliia testowanie zmian.

choooobert committed:
added button to delete all players
changed url of roomService to heroku link

choooobert committed:
removed id from Player interface

choooobert committed:
renamed function from deleteHero to deletePlayer

choooobert committed:
changed POST URI and removed RequestBody

choooobert committed:
Merge branch 'master' into Hubert

choooobert committed:
Merge branch 'master' into Hubert

choooobert committed:
added connection between GameService and Frontend
added enums and interfaces to store data from REST

MJ211 committed:
Added observer for http get which returns players in room
Added redirection in waiting room to game view when max amount of players is reached
Changed some text in HTMLs
Changed redirection from home

MJ211 committed:
Merge branch 'Hubert' of https://github.com/Kamil-Jankowski/ShipsFrontend into Hubert

choooobert committed:
added connection between RandomShipPlacementService and Frontend
added hhtpOptions for every REST method

MJ211 committed:
Improved encapsulation

MJ211 committed:
Merge branch 'Hubert' of https://github.com/Kamil-Jankowski/ShipsFrontend into Hubert

choooobert committed:
added generting new ships when adding new player

MJ211 committed:
Cleaned up the code and added simple rules

choooobert committed:
added connection between shoot-map and GameService

choooobert committed:
removed ship-map and shoot map
moved these components to game
RESTs do not work yet

choooobert committed:
Merge branch 'Hubert' of https://github.com/Kamil-Jankowski/ShipsFrontend into Hubert

Kamil-Jankowski commented on src/app/game/game.component.html:
```diff
@@ -32,7 +46,8 @@
         <p *ngIf="player" >
-          Player {{player.id}}: {{player.name}}
+          Your name: {{player.name}}
+          Turn : {{playerTurn.name}}
```
Jeśli to mamy tutaj, to można usunąć Game info (albo tą linię umieścić w game info)

Kamil-Jankowski commented on src/app/game/game.component.ts:
```diff
@@ -16,33 +23,158 @@ import { MessageService } from '../messages.service';
     public messageService: MessageService,
-  ) { }
+    private notificationService: NotificationService,
+    ) {
+      this.shootNotification = {
+        message: 'Your turn. Shoot!',
+        type: NotificationType.info}
+
+      this.braceNotification = {
+        message: 'Brace yourself!',
+        type: NotificationType.info}
+
+      this.hitNotification = {
+        message: 'Hit!',
+        type: NotificationType.success}
+      
+      this.missNotification = {
+        message: 'Miss!',
+        type: NotificationType.error}
+     }
```
Implementacja powiadomień jest już zmieniona ze względu na tłumaczenia.

Kamil-Jankowski commented on src/app/game/game.component.ts:
```diff
@@ -16,33 +23,158 @@ import { MessageService } from '../messages.service';
+    this.getShipMap();
+
+    for (let i = 1; i <=100; i++) {
+      this.shipMap.push({id : i, status : 0, taken :true});
```
Dlaczego `taken = true`?

Kamil-Jankowski commented on src/app/game/game.component.ts:
```diff
@@ -16,33 +23,158 @@ import { MessageService } from '../messages.service';
+  }
+
+
+  showMessage(message: NotificationMessage) {
```
Do rozważenia, czy tego nie usuniemy

choooobert committed:
added improvements in game component
added logs
changed REST API in order to correspond with backend

Kamil-Jankowski commented:
Przejście do poczekalni można zrobić na kliknięciu bez czekania.

choooobert committed:
improved button status handling

choooobert committed:
added opponent deduction

choooobert committed:
added blocking buttons when opponent's turn
added 3-second-synchronization between frontend and backend
removed messages component

choooobert commented:
Merge request not needed. All changes were merged from branch "Kamil"
